### ISSUE TITLE
LLVM is currently relicensing its code base

### ISSUE THREAD

**Issue by mnme** [include-what-you-use/include-what-you-use#849]
IWYU is currently licensed under LLVM's legacy license, which is being replaced by Apache License v2.0 with LLVM exceptions. Is there interest or are there any efforts of following LLVM and relicensing IWYU?

**Comment by kimgr** [comment]
I haven't really had a chance to think about it.

I'm not sure what the pros and cons and legal ramifications of switching licenses are. As far as I can see, there are no real synergies to IWYU using the same license as LLVM, but I'm probably missing something.

I also did not follow the decision process for when LLVM switched, so I'm not sure what was "wrong" with LLVM's original license. I assume that license is just a set of conditions, and the fact that LLVM no longer uses doesn't make it void.

Thoughts?

**Comment by mnme** [comment]
I mostly wanted to raise awareness that this license, which is in and of itself already pretty "special", is going to be abandoned by LLVM itself. This doesn't mean that it's void or not a good fit for IWYU. I just discovered this fact when I fiddled with the IWYU ArchLinux package.

Maybe this would be a good time to discuss if this license is currently a good fit for the project. But I don't have any preference myself, I'm also fine with closing the issue and not doing anything.

**Comment by aaronpuchert** [comment]
As far as I understand, there were two reasons for the relicensing:

1. parts of the code were under a different license, and so it wasn't easy to move code between them,
2. (particularly corporate) contributors had concerns about patents. The old license didn't say anything about patents, whereas the new license has an implicit but limited patent grant.

But you can read for yourself here: https://llvm.org/foundation/relicensing/.

Since IWYU is pretty niche and I guess it doesn't have any cutting-edge patentable technology, the reasons don't really apply here. Also my understanding is that there is no legal problem with having a different license than LLVM, since Apache 2 is still permissive.

The LLVM foundation did in fact contact all past contributors (or at least commit authors) and ask them to sign off on the relicensing, which doesn't sound like something @kimgr wants to do in his spare time. So I don't want to argue against relicensing, but I think it's probably not worth the effort.

**Comment by kimgr** [comment]
@aaronpuchert Thank you for the educated and thoughtful response.

Good point about getting signoff from past contributors. That's something that might make sense for the LLVM foundation, but not for a guy with a hobby :smile:.

If anything, I would rather like to move toward a simpler 2-clause BSD license. I definitely see IWYU as more of an experimental research project than something that would generate patentable IP.

**Comment by xmnlab** [comment]
hi everyone! just a friendly check if there are any updates about this discussion :)

**Comment by kimgr** [comment]
Not really. I don't think it's likely to happen. Do you have any specific concerns?

**Comment by xmnlab** [comment]
nop actually, just to be aware about any decision. I am trying to build a conda package and I will propose that to conda-forge channel, and I need to specify the license inside the recipe. I need to fill some fields about license, like the license name and the license "family". As LLVM license family is not a valid option there, so for now I just have it as OTHER. but I have the field for the license name as `LLVM Release License`. I hope it would be enough.

thanks for the update.

**Comment by aaronpuchert** [comment]
@xmnlab, the license is called "University of Illinois/NCSA Open Source License", the SPDX identifier is [NCSA](https://spdx.org/licenses/NCSA). "LLVM Release License" is not the actual license name, it simply states that this is the license for LLVM releases. It should probably be changed to "include-what-you-use Release License" or be entirely removed.

**Comment by xmnlab** [comment]
got it, thank you so much for all the explanation. really appreciate that!
